packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp

```diff
@@ -743,8 +743,11 @@ namespace MueLu {
       GetOStream(Runtime0) << "RefMaxwell::compute(): nuking BC nodes of D0" << std::endl;
 
       D0_Matrix_->resumeFill();
-      // Scalar replaceWith = Teuchos::ScalarTraits<SC>::eps();
-      Scalar replaceWith = Teuchos::ScalarTraits<SC>::zero();
+      Scalar replaceWith;
+      if (D0_Matrix_->getRowMap()->lib() == Xpetra::UseEpetra)
+        replaceWith= Teuchos::ScalarTraits<SC>::eps();
+      else
+        replaceWith = Teuchos::ScalarTraits<SC>::zero();
```

Comment:
Why is this necessary?

Reply:
The implicit transpose matrix multiply in Epetra seems to produce an invalid matrix when we have explicit zeros in D0. I have not tried to track this down in Epetra.